Application/EdFi.Ods.Common/Security/Authorization/AuthorizationSegmentEndpoint.cs

```diff
@@ -26,7 +26,7 @@ public AuthorizationSegmentEndpoint(string name, Type type)
         /// <summary>
         /// Gets the name of the rule value (e.g. a property name from the <see cref="EdFiRelationshipsAuthorizationContextData"/>).
         /// </summary>
-        public string Name { get; }
+        public string Name { get; set; }
```

Comment:
Opening this up to "set" is a scary change, classes that don't have setters are meant to be immutable and this change risks breaking that.  We need to figure out how to generate the AuthorizationSegmentEndpoint with the more abstract EducationOrganizationId not just fix it up after.

Reply:
Because of this I moved the logic to AuthorizationBuilder class and create new method 
AddClaimsAuthorizationSegmentList to handle this . Now I am not doing Set for Endpoint name